lib/sources.nix

```diff
@@ -229,6 +229,42 @@ let
 
   canCleanSource = src: src ? _isLibCleanSourceWith || !(pathHasContext (toString src));
 
+  /*
+    Get all files (recursively) that pass a filter function. Unlike other source
+    filtering functions, this will not include empty directories in the final
+    derivation.
+
+    Type: sourceLike -> (String -> Bool) -> Source
+
+    Example:
+      filterFiles ./. (lib.hasSuffix ".html")
+  */
+  filterFiles = src: filter:
+    let
+      includedPaths =
+        builtins.filter
+          filter
+          (lib.filesystem.listFilesRecursive src);
+
+      name =
+        # Fix the problem of store paths names causing a rebuild every time they change.
+        let baseName = builtins.baseNameOf src; in
+        if builtins.match "[a-z0-9]{32}.*" baseName == null
+        then baseName
+        else "source";
+    in
+    cleanSourceWith {
+      inherit name src;
```

Comment:
These two lines can be simplified to [`filter`](https://github.com/NixOS/nixpkgs/pull/112083/files#diff-31011afdfc8a081afccc0785972f74249cf02ea4d10bc41d6b0af992733c9de8R114) after the other changes.